### DISCUSSION TITLE
isSubmitting stops before redirect

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#5586] [Q&A]
When a user clicks to login I show a loading spinner and then on successful login the user is redirected. I don't know if there is something I can do with React Hook Form to prevent this or if it is a next.js auth issue.

Basically what is happening is that the loading spinner stops and the button shows as clickable again for a period of time before the redirect instead of the loading spinner showing the entire time up until the redirect. 

I have uploaded an animated gif to illustrate 

https://gfycat.com/finishedmarvelousliger

And this is the button code

```
              <Button variant="primary" type="submit">
                {isSubmitting ? (
                  <Spinner
                    as="span"
                    animation="border"
                    size="sm"
                    role="status"
                    aria-hidden="true"
                  />
                ) : (
                  "Log in"
                )}
              </Button>
```

**Comment by bluebill1049** [comment]
so the `submitting` state reverted before the `redirect`? can you reproduce in a codesandbox? unmount shouldn't reset the state.

**Comment by ghost** [comment]
Yes, correct. Will be very difficult to do a sandbox because I am using Next.js auth with my own database. However, I think I found a solution or maybe it’s a workaround, not sure. And that was to throw isSubmitSuccessful in there like this:

```
            {isSubmitting || isSubmitSuccessful ? (
              <Button variant="primary" type="submit" disabled>
                <Spinner
                  as="span"
                  animation="border"
                  size="sm"
                  role="status"
                  aria-hidden="true"
                />
                <span className="sr-only"> Submitting ...</span>
              </Button>
            ) : (
              <Button variant="primary" type="submit">
                Submit
              </Button>
            )}
```
